src/shogun/classifier/svm/NewtonSVM.cpp

```diff
@@ -233,9 +220,8 @@ void CNewtonSVM::line_search_linear(float64_t* weights, float64_t* d, float64_t*
 	SGVector<float64_t> outzsv(x_n);
 	SGVector<float64_t> Ysv(x_n);
 	SGVector<float64_t> Xsv(x_n);
-	float64_t t=0.0;
 	SGVector<float64_t> Xd(x_n);
-
+	SGVector<float64_t> weights = get_w();
 	for (int32_t i=0; i<x_n; i++)
 		Xd[i]=features->dense_dot(i, d, x_d);
```

Comment:
@karlnapf making const gives errors here

Reply:
But the definition is
`virtual float64_t dense_dot(int32_t vec_idx1, const float64_t* vec2,
			int32_t vec2_len);`